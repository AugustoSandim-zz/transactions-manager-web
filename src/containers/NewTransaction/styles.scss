@import "../../utils/style/variables.scss";

.new-transaction {
  background-image: url("../../image/bg.svg");
  background-size: 140%;
  background-position: top center;
  background-repeat: no-repeat;
  width: 100%;
  height: 100vh;

  &--back {
    color: white;
    display: block;
    font-size: 18px;
    padding: 20px;
    text-decoration: none;
  }
  
  &__container {
    padding: 20%;

    h1 {
      color: $scale12;
      font-family: "Oswald", sans-serif;
      font-size: 60px;
      width: 200px;
      line-height: 1;
      margin: 80px 0;
    }
    
    input {
      height: 56px;
      margin: 20px 0;
    }

    button {
      float: right;
      margin-top: 24px;
    }
  }
}
