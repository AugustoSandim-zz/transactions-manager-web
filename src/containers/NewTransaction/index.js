import React, {useEffect, useState} from 'react';
import {Link, withRouter} from 'react-router-dom';
import Button from '../../components/Button';
import InputText from '../../components/InputText';
import {currentUser} from '../../services/auth';
import './styles.scss';

function NewTransaction({history}) {
  const [data, setData] = useState({
    transactionValue: '',
    description: '',
    created: '',
  });
  const [errors, setErrors] = useState({
    transactionValue: '',
    transactionType: '',
    description: '',
  });

  const [message, setMessage] = useState('');

  useEffect(() => {
    currentUser();
  }, []);

  const showError = obj => {
    setErrors(obj);
  };

  const handleSubmit = e => {
    e.preventDefault();

    const {transactionValue, description} = data;

    if (!transactionValue.replace(/[^\d]/g, '')) {
      return showError({transactionValue: 'Favor digite um valor ;)'});
    }

    if (!description) {
      return showError({description: 'Digite uma descrição.'});
    }

    try {
      const transactions =
        JSON.parse(localStorage.getItem(currentUser())) || [];
      transactions.push(data);

      localStorage.setItem(currentUser(), JSON.stringify(transactions));

      setMessage('Transação cadastrada com sucesso!');
    } catch (error) {
      return setMessage('Erro ao salvar transação, tente novamente!');
    }
  };

  const handleClick = () => {
    setData({
      ...data,
      created: new Date(),
    });
  };

  const handleCurrencyChange = (e, value, maskedValue) => {
    setData({
      ...data,
      transactionValue: maskedValue,
    });

    setErrors({
      ...errors,
      transactionValue: '',
    });
  };

  const handleChange = e => {
    setData({
      ...data,
      [e.target.name]: e.target.value,
    });

    setErrors({
      ...errors,
      [e.target.name]: '',
    });
  };

  return (
    <div className="new-transaction">
      <Link to="/dashboard" title="Voltar para dashboard" className="new-transaction--back">
        <i className="fa fa-chevron-left" aria-hidden="true" /> Voltar
      </Link>

      <div className="new-transaction__container">
        <h1>Cadastrar nova transação</h1>
        <form onSubmit={handleSubmit}>
          {message && <p className="new-transaction--success">{message}</p>}
          <div>
            <label htmlFor="transactionValue">
              Valor da transação
              <InputText
                currency
                type="currency"
                placeholder="Valor"
                id="transactionValue"
                name="transactionValue"
                onChange={handleCurrencyChange}
                error={errors.transactionValue}
              />
            </label>
          </div>
          <div>
            <label htmlFor="description">
              Descrição
              <InputText
                type="text"
                placeholder="Descrição"
                id="description"
                name="description"
                onChange={handleChange}
                error={errors.description}
              />
            </label>
          </div>

          <Button type="submit" onClick={handleClick}>
            Enviar
          </Button>
        </form>
      </div>
    </div>
  );
}

export default withRouter(NewTransaction);
