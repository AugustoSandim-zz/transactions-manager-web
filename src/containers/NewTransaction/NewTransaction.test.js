/* global expect, jest */
import React from 'react';
import {mount} from 'enzyme';
import {Route, MemoryRouter} from 'react-router-dom';
import NewTransaction from './index';

function renderNewTransaction() {
  return mount(
    <MemoryRouter>
      <Route render={props => <NewTransaction {...props} />} />
    </MemoryRouter>,
  );
}

it('renders without crashing', () => {
  const component = renderNewTransaction();

  expect(component.find('.new-transaction')).toHaveLength(1);
});
