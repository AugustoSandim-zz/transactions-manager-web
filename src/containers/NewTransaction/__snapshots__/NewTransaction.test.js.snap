// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTransaction renders without crashing 1`] = `ReactWrapper {}`;
